src/client/updateClientMetaInfo.js

```diff
@@ -1,64 +1,75 @@
-    if (htmlTag.getAttribute(ssrAttribute) === null) {
-      // initialize tracked changes
-      const addedTags = {}
-      const removedTags = {}
+  // only cache tags for current update
+  const tags = {}
+
+  const htmlTag = getTag(tags, 'html')
+
+  // if this is not a server render, then update
+  if (htmlTag.getAttribute(ssrAttribute) === null) {
+    // initialize tracked changes
+    const addedTags = {}
+    const removedTags = {}
+
+    Object.keys(newInfo).forEach((type) => {
+      // ignore these
+      if ([
+        'titleChunk',
+        'titleTemplate',
+        'changed',
+        '__dangerouslyDisableSanitizers',
+        '__dangerouslyDisableSanitizersByTagID'
+      ].includes(type)) {
+        return
+      }
 
-      Object.keys(newInfo).forEach((key) => {
-        switch (key) {
-          // update the title
-          case 'title':
-            updateTitle(options)(newInfo.title)
-            break
-          // update attributes
-          case 'htmlAttrs':
-            updateTagAttributes(options)(newInfo[key], htmlTag)
-            break
-          case 'bodyAttrs':
-            updateTagAttributes(options)(newInfo[key], document.getElementsByTagName('body')[0])
-            break
-          case 'headAttrs':
-            updateTagAttributes(options)(newInfo[key], document.getElementsByTagName('head')[0])
-            break
-          // ignore these
-          case 'titleChunk':
-          case 'titleTemplate':
-          case 'changed':
-          case '__dangerouslyDisableSanitizers':
-            break
-          // catch-all update tags
-          default:
-            const headTag = document.getElementsByTagName('head')[0]
-            const bodyTag = document.getElementsByTagName('body')[0]
-            const { oldTags, newTags } = updateTags(options)(key, newInfo[key], headTag, bodyTag)
-            if (newTags.length) {
-              addedTags[key] = newTags
-              removedTags[key] = oldTags
-            }
-        }
-      })
+      if (type === 'title') {
+        // update the title
+        updateTitle(newInfo.title)
+        return
+      }
 
-      // emit "event" with new info
-      if (typeof newInfo.changed === 'function') {
-        newInfo.changed.call(this, newInfo, addedTags, removedTags)
+      if (['htmlAttrs', 'headAttrs', 'bodyAttrs'].includes(type)) {
+        const tagName = type.substr(0, 4)
+        updateAttribute(options, newInfo[type], getTag(tags, tagName))
+        return
       }
-    } else {
-      // remove the server render attribute so we can update on changes
-      htmlTag.removeAttribute(ssrAttribute)
-    }
+
+      const { oldTags, newTags } = updateTag(
+        options,
+        type,
+        newInfo[type],
+        getTag(tags, 'head'),
+        getTag(tags, 'body')
+      )
+
+      if (newTags.length) {
+        addedTags[type] = newTags
+        removedTags[type] = oldTags
+      }
+    })
+
+    return { addedTags, removedTags }
```

Comment:
I think there is still a bug or documentation issue here, the oldTags variable gets added to removedTags but old !== removed. I left some todo comments in the test about this as well, as fixing this could be a breaking change (even though the docs also explictly state removedTags). See eg here: https://github.com/pimlie/vue-meta/blob/feat-jest/test/utils/meta-info-data.js#L74